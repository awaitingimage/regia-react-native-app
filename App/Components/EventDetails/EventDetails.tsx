import * as React from "react";
import { Text, View } from "react-native";
import { Event } from "../../Lib/Events";
import styles from "./Style";

interface Props {
  event: Event;
}

const EventDetails: React.SFC<Props> = ({ event }: Props) => {
  return (
  <View style={styles.container}>
    <Text style={styles.body}>{event.title}</Text>
  </View>
); };

export default EventDetails;
