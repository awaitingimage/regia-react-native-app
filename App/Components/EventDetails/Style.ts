import { StyleSheet } from "react-native";
import { Colors, Fonts, Metrics } from "../../Themes/index";

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
    paddingHorizontal: Metrics.baseMargin,
    paddingVertical: Metrics.baseMargin * 2,
  },
  body: {
    ...Fonts.style.description,
    color: Colors.coal,
    paddingBottom: Metrics.baseMargin * 3,

  }
});
