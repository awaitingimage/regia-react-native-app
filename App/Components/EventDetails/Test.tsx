/// <reference types="@types/jest" />
import * as React from "react";
import { Text, View } from "react-native";
import * as renderer from "react-test-renderer";
import EventDetails from "./EventDetails";

const testEvent =  {
  title: "Sailing in Cardiff Bay",
  publicPrivate: "private",
  startDate: "2018-02-03",
  endDate: "2018-02-04",
  dateline: "",
  datelineEra: "unknown",
  datelineAOGuide: "N/A",
  type: "Maritime Weekend",
  nationalShow: true,
  localTraining: false,
  nonRegiaEvent: false,
  cancelled: false,
  address1: "Cardiff",
  address2: "",
  postcode: "",
  country: "United Kingdom",
  website: "",
  FacebookLink: "",
  details: "Sailing event, liaise with Maritime Officer if interested"
};

test("EventDetails component renders correctly", () => {
  const tree = renderer.create(
  <EventDetails
    event={testEvent}
  />,
).toJSON();
  expect(tree).toMatchSnapshot();
});
