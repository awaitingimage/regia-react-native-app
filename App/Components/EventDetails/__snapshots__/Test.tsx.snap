// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#2d2d2d",
        "fontFamily": "Avenir-Book",
        "fontSize": 14,
        "paddingBottom": 30,
      }
    }
  >
    Sailing in Cardiff Bay
  </Text>
</View>
`;
